### PR TITLE
DEV-104: Make AI responses warmer and clearer

### PR DESCRIPTION
**Pull request by irachrist1**
## What changed

- Reworked the shared Daylens voice contract so every AI surface uses warm, everyday language, short sentences, and the amount of detail the question needs.
- Kept the grounding contract intact: no invented facts, empty praise, judgment, or fake familiarity.
- Added a dedicated conversational greeting path, so prompts such as “Hey? how’s it going?” get a brief natural model-written reply instead of the resolver capability menu.
- Softened the unsupported-question fallback and removed conflicting “dry” instructions from daily, weekly, and monthly wrap prompts.
- Updated `PRODUCT.md` and the AI, Apps, Timeline, briefs/wraps, onboarding, Settings, and work-memory specs to share the same clarity rules.
- Added regression coverage for greeting classification, shared prompt coverage, and the warm/clear voice contract.
- Removed the stale `recharts` manual chunk left behind after the chart dependency was deleted, restoring clean standalone renderer builds on Windows and Linux CI.

## Why

The AI architecture was grounded, but the writing contract still optimized for a dry, forensic tone. The greeting in the issue screenshot was not recognized as conversation, so it fell into a generic list of capabilities. This fixes both at the shared contract and routing level.

The platform runtime checks also exposed a separate cleanup remnant: Vite still tried to create a `recharts` chunk even though the app no longer imports or depends on `recharts`.

## User impact

Daylens now sounds more human and is easier to understand for technical and non-technical users, without becoming chatty, motivational, or less trustworthy. Clean production renderer builds work again on every platform.

## Validation

- `npm run typecheck`
- `npm run lint` — 0 errors; existing unrelated warnings remain
- `npm test` — 107 files, 674 passed, 0 failed, 1 skipped
- `npm run build:renderer`
- Real Electron app: “Hey? how’s it going?” → “Hey—doing well, thanks for asking. What’s on your mind today?”
- Real Electron app: thin-day query stayed short, specific, and honest about the limited captured evidence

Linear: DEV-104

---

> [!NOTE]
> **Medium Risk**
> Changes central chat routing and every surface that imports `VOICE_SYSTEM_PROMPT`; greetings now depend on a live API call instead of deterministic text, though grounding invariants are preserved.
> 
> **Overview**
> **Shared voice** shifts from dry, forensic tone to **warm, everyday language** with short sentences and detail matched to the question, while keeping the evidence and citation rules unchanged (`voiceContract.ts`, wrap/brief prompts).
> 
> **Chat routing** adds `smallTalk.ts` to detect pure greetings/check-ins (without swallowing real questions) and answer them via the **Settings model** and a dedicated system prompt—not a fixed string or resolver capability menu. Unsupported planner questions get a softer boundary line in `planner.ts`.
> 
> **Docs** (`PRODUCT.md`, `ai.md`, apps/timeline/briefs/onboarding/settings/work-memory specs) now describe the same clarity rules, including a new conversational-turn invariant for casual hellos.
> 
> **Build:** removes the unused `vendor-charts` / `recharts` manual chunk from `vite.renderer.config.ts` after the dependency was removed.
> 
> **Tests** cover greeting classification, voice prompt warmth, and `smallTalk` in the curated prompt-site list.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f1e1e30694caee583cdc14b96304e6196875d9ac. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Conversational greeting detection now triggers dynamic AI-generated responses instead of fixed messages

* **Documentation**
  * Updated voice specifications across all features emphasizing warmth, clarity, and everyday language
  * Refined guidance on explaining technical terms and maintaining consistent tone

* **Tests**
  * Added test coverage for greeting detection and voice specification requirements

* **Chores**
  * Simplified build configuration

### CODE REVIEW

File: src/main/ai/smallTalk.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { VOICE_SYSTEM_PROMPT } from './voiceContract'
+
+const SIMPLE_GREETING_RE =
+  /^\s*(?:hey|hi|hello|howdy|hiya|helo|yo|sup|good\s+(?:morning|afternoon|evening)|test|testing)\s*[!?.]*\s*$/i
+
+const CHECK_IN_RE =
+  /^\s*(?:how(?:'s| is) it going|how are you|what(?:'s| is) up|how(?:'s| is) your day(?: going)?)\s*[!?.]*\s*$/i
+
+const GREETING_WITH_CHECK_IN_RE =
+  /^\s*(?:hey|hi|hello|howdy|hiya|helo|yo|sup|good\s+(?:morning|afternoon|evening))[\s,!?.—-]+(?:how(?:'s| is) it going|how are you|what(?:'s| is) up|how(?:'s| is) your day(?: going)?)\s*[!?.]*\s*$/i
```
[the remark this answers is not in the document]
Comment (irachrist1) [reply src/main/ai/smallTalk.ts]: Fixed in f1e1e30. Both greeting regexes now accept ASCII and smart apostrophes, with regression cases for “How’s your day?”, “how’s it going?”, and “what’s up?”. Focused tests, typecheck, and lint pass.

**Comment by irachrist1** [comment]
@coderabbitai review

Follow-up: fixed the smart-apostrophe greeting edge case and added regression coverage in f1e1e30. I also incorporated the optional sentence-variety suggestion in the AI spec. The docstring-coverage warning is intentionally not applied: this TypeScript codebase does not use docstring coverage as a project gate, and adding boilerplate to internal regex constants would not improve the public contract or runtime clarity.
